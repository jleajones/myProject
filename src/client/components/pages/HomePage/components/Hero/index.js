import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import Container from '@material-ui/core/Container';
import makeStyles from '@material-ui/core/styles/makeStyles';

const useStyles = makeStyles(theme => ({
  paper: {
    padding: theme.spacing(2),
    color: theme.palette.text.secondary
  }
}));

const Hero = ({ users }) => {
  const classes = useStyles();
  const keys = users.length ? Object.keys(users[0]) : [];
  return (
    <Container>
      <Grid container spacing={2}>
        {users.map(user => (
          <Grid item key={user.id} xs={12}>
            <Paper className={classes.paper}>
              {keys.map(key => (
                <Typography key={`${user.id}-${key}`}>
                  <b>{key}</b> {user[key]}
                </Typography>
              ))}
            </Paper>
          </Grid>
        ))}
      </Grid>
    </Container>
  );
};

Hero.propTypes = {
  users: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Hero;
