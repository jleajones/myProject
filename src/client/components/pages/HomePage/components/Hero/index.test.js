import React from 'react';
import { shallow } from 'enzyme';
import Hero from '.';

describe('<Hero />', () => {
  test('Renders component without crashing', () => {
    const component = shallow(<Hero />);

    expect(component).toMatchSnapshot();
  });
});
