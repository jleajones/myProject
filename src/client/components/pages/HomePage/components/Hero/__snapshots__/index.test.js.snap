// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> Renders component without crashing 1`] = `
<WithStyles(ForwardRef(Container))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="1"
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-1"
      >
        <WithStyles(ForwardRef(Typography))
          key="1-id"
        >
          <b>
            id
          </b>
           
          1
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="1-username"
        >
          <b>
            username
          </b>
           
          HarveyD
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="1-email"
        >
          <b>
            email
          </b>
           
          harvey@email.com
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="1-joined"
        >
          <b>
            joined
          </b>
           
          2019-07-07T04:45:04.524Z
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="1-last_logged_in"
        >
          <b>
            last_logged_in
          </b>
           
          2019-07-07T04:45:04.524Z
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="3"
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-1"
      >
        <WithStyles(ForwardRef(Typography))
          key="3-id"
        >
          <b>
            id
          </b>
           
          3
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="3-username"
        >
          <b>
            username
          </b>
           
          jleajones
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="3-email"
        >
          <b>
            email
          </b>
           
          jleajones@email.com
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="3-joined"
        >
          <b>
            joined
          </b>
           
          2019-08-12T19:24:21.942Z
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="3-last_logged_in"
        >
          <b>
            last_logged_in
          </b>
           
          2019-08-12T19:24:21.942Z
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      key="4"
      xs={12}
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-paper-1"
      >
        <WithStyles(ForwardRef(Typography))
          key="4-id"
        >
          <b>
            id
          </b>
           
          4
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="4-username"
        >
          <b>
            username
          </b>
           
          darmstr2
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="4-email"
        >
          <b>
            email
          </b>
           
          darmstr2@email.com
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="4-joined"
        >
          <b>
            joined
          </b>
           
          2019-08-12T19:24:39.744Z
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          key="4-last_logged_in"
        >
          <b>
            last_logged_in
          </b>
           
          2019-08-12T19:24:39.744Z
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Container))>
`;
